pkg/resources/cruisecontrol/configmap.go

```diff
@@ -145,10 +146,6 @@ func GenerateCapacityConfig(kafkaCluster *v1beta1.KafkaCluster, log logr.Logger,
 	// If there is already a config added manually, use that one
 	if kafkaCluster.Spec.CruiseControlConfig.CapacityConfig != "" {
 		userProvidedCapacityConfig := kafkaCluster.Spec.CruiseControlConfig.CapacityConfig
-		updatedProvidedCapacityConfig := ensureContainsDefaultBrokerCapacity([]byte(userProvidedCapacityConfig), log)
```

Comment:
Please ensure that there is config added for those brokers that the user missed to provide a CC config for.
E.g.: if there are 3 brokers and the user provided only CC capacity config for one broker than config is added for the remaining two brokers. Also add a unit test for this scenario.

Reply:
It is ensured, if it's in kafkacluster.Status.BrokerState it will receive a CC config just not a default -1 indexed one but one specifically assigned to it. I'll create a unit test as you requested.

Reply:
It is not ensured right now, since when the user provides the capacity config, we will simply return in line 152. But I am not sure that is the right way to do, to silently appending additional configurations to this very important config. I know earlier we always injected the default value, but that can lead to not so precise CC plans. I am leaning towards a solution where we are checking if the user provided all the info required and if not just failing with an error.

Reply:
That would be a change in the functionality and needs to be tracked in the release notes for those who upgrade and prior provided a cc config. 

I think that we can provide a good enough capacity values for the missing brokers since CPU and storage can be computed automatically. Only the NWIN and NWOUT would be a default value.

Reply:
Oh hell, you're right, sorry it is not ensured there. I misunderstood it a bit. What should happen here? I kinda understand the idea of Bálint with the error so that the user knows that the config they gave is not a complete one but at the same time won't it be annoying if for example all they care about is setting the config for one broker and they're fine with a default for the rest?

Reply:
We agreed offline to implement the following:
If the user adds capacity config, but not for every broker we shall add values automatically for every missing broker.
If the user adds capacity config, but with a -1 default value we shall use that, since the user technically specifically asked for their default and we shouldn't change something like that behind the scenes.
If the user adds capacity config for every broker then that's fine as well, that will be used.

Reply:
Okay, so hopefully the latest commit should fix this issue and do what I described in the previous comment.